src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int RESIZE_MULTIPLIER = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] values;
+    private int size;
+
+    public MyHashMap() {
+        values = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        ensureCapacity();
+        Node<K, V> node = new Node<>(key, value);
+        int position = position(node);
+        if (values[position] == null) {
+            values[position] = node;
+            size++;
+        } else {
+            if (values[position].put(node)) {
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int position = position(key);
+        if (values[position] == null) {
+            return null;
+        } else {
+            return values[position].get(key);
+        }
```

Comment:
looks like "else" is not needed here
```suggestion
        } 
        return values[position].get(key);
        }
```